File: 建造者模式/food.py
from pack import Wrapper, Bottle
from Item import Item

class Burger(Item):
  def Pack(self):
    return Wrapper().Pack() 

class VegBurger(Burger):
  def __init__(self):
    self.name = "Veg Burger"
    self.price = 1.5

class ChickenBurger(Burger):
  def __init__(self):
    self.name = "Chicken Burger"
    self.price = 2.5

class ClodDrink(Item):
  def Pack(self):
    return Bottle().Pack() 

class Coke(ClodDrink):
  def __init__(self):
    self.name = "Coke"
    self.price = 0.9

class Pepsi(ClodDrink):
  def __init__(self):
    self.name = "Pepsi"
    self.price = 0.4

File: 建造者模式/food.go
package main

type Burger struct {
}

func (c *Burger) Pack() string {
	packing := Wrapper{}
	return packing.Pack()
}

type VegBurger struct {
	Burger
}

func (v *VegBurger) Price() float64 {
	return float64(3.8)
}
func (v *VegBurger) Name() string {
	return "VegBurger"
}

type ChickenBurger struct {
	Burger
}

func (c *ChickenBurger) Price() float64 {
	return float64(2.8)
}

func (c *ChickenBurger) Name() string {
	return "ChickenBurger"
}

type ColdDrink struct {
}

func (c *ColdDrink) Pack() string {
	packing := Bottle{}
	return packing.Pack()
}

type Pepsi struct {
	ColdDrink
}

func (c *Pepsi) Price() float64 {
	return float64(1.8)
}

func (c *Pepsi) Name() string {
	return "Pepsi"
}

type Coke struct {
	ColdDrink
}

func (c *Coke) Price() float64 {
	return float64(0.8)
}

func (c *Coke) Name() string {
	return "Coke"
}
